api.go

```diff
@@ -174,10 +174,12 @@ func newAPI(cli httpClient) APIInterface {
 
 func (impl *apiImpl) CreateAPI(ctx context.Context, api *API, opts *ResourceCreateOptions) (*API, error) {
 	var createdAPI API
-
+	if !ensureClusterID(impl.client, opts) {
+		return nil, ErrClusterIDNotExist
+	}
 	appID := opts.Application.ID
 	uri := path.Join(_apiPathPrefix, "apps", appID.String(), "apis")
-	err := impl.client.sendPostRequest(ctx, uri, "", api, jsonPayloadDecodeFactory(&createdAPI))
+	err := impl.client.sendPostRequest(ctx, uri, "", api, jsonPayloadDecodeFactory(&createdAPI), appendHeader(mapClusterIdFromHttpClient(impl.client), mapClusterIdFromOpts(opts)))
```

Comment:
We don't need this: `mapClusterIdFromHttpClient(impl.client)`. How to choose the Cluster ID should be done by the `httpClient`. Use the one in storage or, overriding from the given one (from opts).

Reply:
Removed